iceoryx_hoofs/source/posix_wrapper/named_pipe.cpp

```diff
@@ -94,15 +96,48 @@ NamedPipe::NamedPipe(const IpcChannelName_t& name,
 
     if (sharedMemory.has_error())
     {
-        std::cerr << "Unable to open shared memory: \"" << convertName(name) << "\" for named pipe \"" << name << "\""
-                  << std::endl;
+        std::cerr << "Unable to open shared memory: \"" << convertName(NAMED_PIPE_PREFIX, name)
+                  << "\" for named pipe \"" << name << "\"" << std::endl;
         m_isInitialized = false;
         m_errorValue = (channelSide == IpcChannelSide::CLIENT) ? IpcChannelError::NO_SUCH_CHANNEL
                                                                : IpcChannelError::INTERNAL_LOGIC_ERROR;
         return;
     }
 
-    m_sharedMemory.emplace(std::move(sharedMemory.value()));
+    cxx::expected<Semaphore, SemaphoreError> sendSemaphore = cxx::error<SemaphoreError>(SemaphoreError::UNDEFINED);
+    cxx::expected<Semaphore, SemaphoreError> receiveSemaphore = cxx::error<SemaphoreError>(SemaphoreError::UNDEFINED);
+    if (sharedMemory->hasOwnership())
+    {
+        sendSemaphore = Semaphore::create(CreateNamedSemaphore,
+                                          convertName(SEND_SEMAPHORE_PREFIX, name).c_str(),
+                                          static_cast<mode_t>(S_IRUSR | S_IWUSR),
+                                          static_cast<unsigned int>(maxMsgNumber));
+        receiveSemaphore = Semaphore::create(CreateNamedSemaphore,
+                                             convertName(RECEIVE_SEMAPHORE_PREFIX, name).c_str(),
+                                             static_cast<mode_t>(S_IRUSR | S_IWUSR),
+                                             0U);
+    }
```

Comment:
Do we need some additional handling for obsolete semaphores after an application crashed? Like cleaning them up once we successfully created the shm?